apps/browser/qml/pages/components/TabView.qml

```diff
@@ -40,131 +34,230 @@ SilicaGridView {
     signal closeAllCanceled
     signal closeAllPending
 
-    onCountChanged: if (count > 0) closingAllTabs = false
-    onClosingAllTabsChanged: if (closingAllTabs) closeAllPending()
-
-    width: parent.width - Theme.horizontalPageMargin
-    height: parent.height
-    x: Theme.horizontalPageMargin
-    currentIndex: -1
-    header: PageHeader {
-        //: Tabs
-        //% "Tabs"
-        title: qsTrId("sailfish_browser-he-tabs")
-    }
-    footer: spacer
-    cellHeight: thumbnailHeight + Theme.paddingLarge
-    cellWidth: thumbnailWidth + Theme.paddingLarge
+    property var _remorsePopup
 
-    delegate: TabItem {
-        id: tabItem
+    // TODO: Delete height calculation for Pulley Menu after adding popup menu
+    property real _pulleyMenuApparentHeight
+    readonly property int _pulleyMenuHeight: 7 * Theme.pixelRatio
+    function _changeMenuApparentHeight(contentY) {
+        _pulleyMenuApparentHeight = Math.min(Math.round(_pulleyMenuHeight - tabs.tabBarHeight - contentY), _pulleyMenuHeight)
+    }
 
-        enabled: !closingAllTabs
-        opacity: enabled ? 1.0 : 0.0
-        Behavior on opacity { FadeAnimator {}}
+    anchors.fill: parent
 
-        width: thumbnailWidth
-        height: thumbnailHeight
+    Private.TabView {
+        id: tabs
 
-        GridView.onAdd: AddAnimation {
-            target: tabItem
+        anchors {
+            fill: parent
+            bottomMargin: tabsToolBar.height
         }
-        GridView.onRemove: RemoveAnimation {
-            target: tabItem
+
+        header: Private.TabBar {
+            model: headerModel
+            Rectangle {
+                anchors {
+                    fill: parent
+                    topMargin: _pulleyMenuApparentHeight
```

Comment:
I really like the way you've handled hiding the top menu highlight; it's very ingenious. However it seems to cause some issues, because `_pulleyMenuApparentHeight` doesn't update when the normal/private tab changes. I tried calling `_changeMenuApparentHeight()` when the `currentIndex` changed, but that resulted in other animation artefacts.

Perhaps as an alternative consider adding in a covering bar to the two `Private.TabItem` items instead. This way the bar will slide left/right when the grids slide. Something like this (just a rough idea):
```
Rectangle {
    color: Theme.colorScheme == Theme.LightOnDark ? "black" : "white"
    width: parent.width
    height: Math.max(0, _pulleyMenuHeight - y)
    y: Math.max(0, _pulleyMenuHeight - _persistentTabView.contentY + _persistentTabView.originY)
}   
```
It makes the code a bit messy, but then this is only a temporary solution until the menu is changed anyway.

Reply:
Thank you, so much better. done

Reply:
Thanks @xlise. Unfortunately I don't see the change in practice (did you push it to the repo? I see a force-push, but I don't see any changes; it looks like a rebase).

Reply:
Sorry, I haven't added any files. done

Reply:
Thank you! You should be safe to remove the `_changeMenuApparentHeight()` logic now, and replace the header topMargin to be a fixed height. I'll add a comment about it above.